webfrontend/src/components/auth/register/register.tsx

```diff
@@ -0,0 +1,271 @@
+import { useCallback, useState, FormEvent, useEffect, useRef } from "react";
+import { postRequestContent } from "../../../utils/request";
+import { triggerErrorMessage } from "../../../utils/locals";
+import { isAlphaNumeric } from "../../../utils/primitives";
+import { getURLParams } from "../../../utils/url";
+import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { useSelector } from "react-redux";
+import { login } from "../../../redux/auth-slice";
+import { RootState } from "../../../redux/store";
+
+export const Register = (): JSX.Element => {
+    const [name, setName] = useState<string>('');
+    const [username, setUsername] = useState<string>('');
+    const [password, setPassword] = useState<string>('');
+    const [password2, setPassword2] = useState<string>('');
+    const [isChecked, setIsChecked] = useState<boolean>(false);
+
+    const [nameWarning, setNameWarning] = useState<string>('');
+    const [usernameWarning, setUsernameWarning] = useState<string>('');
+    const [passwordWarning, setPasswordWarning] = useState<string>('');
+    const [password2Warning, setPassword2Warning] = useState<string>('');
+    const [checkboxWarning, setCheckboxWarning] = useState<string>('');
+
+    const [bottomErrorMessage, setBottomErrorMessage] = useState<string>('');
+
+    const navigate = useNavigate();
+
+    const dispatch = useDispatch();
+    const auth = useSelector((state: RootState) => state.auth);
+
+    // useRef is used to ensure states are synced before form submission,
+    const passwordRef = useRef<string>();
+    const password2Ref = useRef<string>();
+    const checkboxRef = useRef<boolean>();
+
+    passwordRef.current = password;
+    password2Ref.current = password2;
+    checkboxRef.current = isChecked;
+
+    useEffect(() => {
+        if (auth.isVerified && auth.isLoggedIn) {
+            const { next } = getURLParams();
+            navigate(next ? next : '/');
+        }
+    }, [auth, navigate]);
+
+    function IsUniqueUsername() {
+        fetch('/check_unique_username_async?username=' + username, {
```

Comment:
Please make use of the `useCallback` hooks instead of declaring nested functions.